routing/pathfind.go

```diff
@@ -1277,6 +1281,15 @@ func processNodeForBlindedPath(g Graph, node route.Vertex,
 		return nil, false, nil
 	}
 
+	// If we have explicity been told to consider this node as the 
+	// penultimate hope.
+	if len(restrictions.nodePenultimateSet) > 0 {
```

Comment:
use `!restrictions.nodePenultimateSet.IsEmpty()` once https://github.com/lightningnetwork/lnd/pull/9159 is merged